### PR TITLE
Add checkedChildren and uncheckedChildren props to Choice

### PR DESCRIPTION
**Pull request by sawyerh**
### Added

Added a `checkedChildren` and `uncheckedChildren` prop to the `Choice` component. These props conditionally display content below the Choice field when the field is checked or unchecked, respectively. 

No styling is add to the children, as there are many use cases for this, so I didn't want to add any constraints. Here's an example of one potential use case:

checkedchildren

This is a prerequisite for building out the [Exposed Within form pattern](https://jira.cms.gov/browse/HDSG-150).

----

### Note for reviewers

**I'd suggest viewing the diff with whitespaced ignore. You can [view that diff here](https://github.com/CMSgov/design-system/pull/173/files?w=1).**

### CODE REVIEW

File: packages/core/src/components/ChoiceList/Choice.jsx [hunk 1]
```diff
@@ -11,41 +11,88 @@ import uniqueId from 'lodash.uniqueid';
  * to the `input` element, so you can use this to set additional attributes if
  * necessary.
  */
-export const Choice = function(props) {
```
Comment (sawyerh) [inline packages/core/src/components/ChoiceList/Choice.jsx]: As I mention in the PR description, an easier way to view the diff for this file is to view the diff with whitespace ignored. You can view that diff here: https://github.com/CMSgov/design-system/pull/173/files?w=1

File: packages/core/src/components/ChoiceList/Choice.test.jsx [hunk 2]
```diff
@@ -3,22 +3,32 @@ import FormLabel from '../FormLabel/FormLabel';
 import React from 'react';
 import { shallow } from 'enzyme';
 
-describe('Choice', () => {
-  const label = 'George Washington';
+const label = 'George Washington';
+
+function shallowRender(customProps = {}, children = label) {
```
Comment (sawyerh) [inline packages/core/src/components/ChoiceList/Choice.test.jsx]: Added this to make the other tests a lot more DRY.

File: packages/core/src/components/ChoiceList/Choice.test.jsx [hunk 3]
```diff
@@ -31,200 +41,125 @@ describe('Choice', () => {
   });
 
   it('is not checked', () => {
-    const props = {
```
Comment (sawyerh) [inline packages/core/src/components/ChoiceList/Choice.test.jsx]: You can ignore these changes. This is just switching over to use the DRY'er `shallowRender` method. Skip down to the next comment where I point out what's actually new.

File: packages/core/src/components/ChoiceList/Choice.test.jsx [hunk 4]
```diff
@@ -274,35 +209,167 @@ describe('Choice', () => {
     ).toBe(inputBId);
   });
 
+  describe('state', () => {
```
Comment (sawyerh) [inline packages/core/src/components/ChoiceList/Choice.test.jsx]: 👋 New tests below this line.

File: packages/core/src/components/ChoiceList/Choice.jsx [hunk 5]
```diff
@@ -11,41 +11,88 @@ import uniqueId from 'lodash.uniqueid';
...
+    }
+
+    if (!this.isControlled) {
+      this.setState({ checked: evt.target.checked });
+    }
+  }
+
+  render() {
+    /* eslint-disable prefer-const */
+    let {
```
Comment (pwolfert) [inline packages/core/src/components/ChoiceList/Choice.jsx]: It seems a shame for all of this to be declared with `let` when only `id` is changed.

File: packages/core/src/components/ChoiceList/Choice.test.jsx [hunk 6]
```diff
@@ -3,22 +3,32 @@ import FormLabel from '../FormLabel/FormLabel';
...
-      name: 'presidents',
-      value: label
-    };
-    const wrapper = shallow(
-      <Choice {...props}>
-        <p>
-          <strong>Hello</strong> World
-        </p>
-      </Choice>
+    const data = shallowRender(
```
Comment (pwolfert) [inline packages/core/src/components/ChoiceList/Choice.test.jsx]: Could say `const { wrapper } = ` to keep the other lines the same.

Comment (sawyerh) [reply packages/core/src/components/ChoiceList/Choice.test.jsx]: oh nice, good point

**Pull request merged**